### PR TITLE
Load applications from all subgraphs

### LINKED ISSUE #534: multichain: on project details, load round applications from all the supported subgraphs
**User Story**
As a project owner, 
I want to see all of the rounds I have applied to with a given project regardless of network 
so that I can have an overview of my round applications 

**Acceptance Criteria**
GIVEN that I am on the Project Details page
WHEN I look at my applications 
THEN I see all of my applications across all supported networks (not just the ones on the network I am currently connected to)

**Notes**
- If I want to see my applications to all rounds for a project, today we are only showing you your applications from the current network (i.e. the one your wallet is connected to) 

**Designs**
* [Figma](https://www.figma.com/file/obBzCCL8KwoTlVBb5g3ccN/Grant-Hub---Wireframes?node-id=3112%3A21936)

**Engineering Notes**

### PR DESCRIPTION
**Pull request by DanieleSalatti**
Closes #534

### CODE REVIEW

File: client/src/actions/projects.ts [hunk 1]
```diff
@@ -309,77 +305,93 @@ export const fetchApplicationStatusesFromContract =
...
-      chain,
-      projectID
-    );
+    const { web3Provider } = global;
+
+    await web3Provider?.chains?.forEach(async (chain) => {
+      const projectApplicationID = generateUniqueRoundApplicationID(
+        chain.id,
+        projectID
+      );
```
Comment (gravityblast) [inline client/src/actions/projects.ts:339]: @DanieleSalatti the id for the application is unique per project hashing `chainID,  projectRegistryAddress, projectID` (using the registry address where the project has been created) but with this code we have a different one based on the round chain

Comment (DanieleSalatti) [reply client/src/actions/projects.ts:339]: Good catch!

We make the same mistake when we send the application  (that's why it works). Will fix that as part of this PR.

Comment (gravityblast) [reply client/src/actions/projects.ts:339]: yeah in theory for now everything was on the same network so this starts having problem when we are really multi chain

File: client/src/actions/projects.ts [hunk 2]
```diff
@@ -309,77 +305,95 @@ export const fetchApplicationStatusesFromContract =
             status: projectApplication.status,
           });
         }
-      } catch (error) {
+      } catch (error: any) {
         datadogRum.addError(error, { roundAddress });
         console.error("getApplicationStatusFromContract() error", {
```
Comment (codenamejason) [inline client/src/actions/projects.ts:310]: @DanieleSalatti do we need a console.error here if we report to ddog and dispatch the error?

File: client/src/actions/roundApplication.ts [hunk 3]
```diff
@@ -270,7 +275,9 @@ export const submitApplication =
         roundAddress,
         projectId: projectID,
       });
-      dispatch<any>(fetchProjectApplications(projectID, chainID));
+      dispatch<any>(
+        fetchProjectApplications(projectID, Number(projectChainId), process.env)
+      );
     } catch (e) {
       datadogRum.addError(e);
       console.error("error calling applyToRound:", e);
```
Comment (codenamejason) [inline client/src/actions/roundApplication.ts:283]: also here we could remove the console.error no? Also on a side note, when we dispatch the applicationError(li 285) we don't actually send the actual error, just the status and generic message.

File: client/src/components/grants/ApplicationCard.tsx [hunk 4]
```diff
@@ -85,11 +95,19 @@ export default function ApplicationCard({
       borderWidth="1px"
       borderRadius="md"
     >
-      <Box p={2} mb={4}>
+      <Box p={2} mb={1}>
         <span className="text-[16px] text-gitcoin-gray-400">
           {props.round?.programName}
         </span>
       </Box>
+      <div className="flex justify-end w-fit mr-1 align-middle mb-2">
```
Comment (codenamejason) [inline client/src/components/grants/ApplicationCard.tsx:103]: we may have to remove this and do on another ticket after design has approved the look/feel of the icon location/placement etc..

Looks good from Eng. standpoint 😁

**Review by codenamejason — APPROVED** [review]
Looks Good 🙌 left a couple comments

**Review by michellema1208 — APPROVED** [review]

File: client/src/actions/projects.ts (hunk 2 above)
Comment (DanieleSalatti) [reply client/src/actions/projects.ts:310]: Replacing with a datadog logger

File: client/src/actions/roundApplication.ts (hunk 3 above)
Comment (DanieleSalatti) [reply client/src/actions/roundApplication.ts:283]: Refactoring a bit

**Pull request merged**